packages/components/Icon/src/Icon.types.ts

```diff
@@ -0,0 +1,35 @@
+import { SvgProps } from 'react-native-svg';
+export const iconName = 'Icon';
+
+export interface IFontIconProps
+{
+  fontFamily?: string;
+  fontSrcFile?: string;
+  codepoint: number;
+  fontSize?: number;
+  color?: string;
+}
+
+export interface ISvgIconProps
+{
+  uri?: string;
+  src?: React.FC<SvgProps>;
+  viewBox?: string;
+  color?: string;
+}
+
+export interface IRasterImageIconProps
+{
+  src: object;
+}
+
+export interface IIconProps
+{
+  ariaLabel?: string;
+  fontSource?: IFontIconProps;
```

Comment:
These interfaces aren't really suitable for an icon control, it is too hard to use. Usually one would expect some sort of name -> values lookup to happen as part of the icon. If we want this interface I might call this object `IconBase` or something that denotes it is a non-opinionated component.

Reply:
> Usually one would expect some sort of name -> values lookup to happen as part of the icon.

Can you elaborate on what you mean by this? Icons referenced by name is for the second mile stone. This milestone is about getting font icon drawing mechanism.

Reply:
When we talked about this in the call yesterday you mentioned putting properties of all three types into a single object so a single object can be passed through slots. But in this model, couldnt the single object be an IIconProps object?